tests/integration_tests/test_visualization.py

```diff
@@ -326,6 +330,94 @@ def test_visualization_compare_classifiers_from_prob_csv_output_saved(
             print("Error: %s - %s." % (e.filename, e.strerror))
 
 
+def test_visualization_compare_classifiers_from_prob_csv_output_saved_api(
+        csv_filename
+):
+    """Ensure pdf and png figures from the experiments can be saved.
+
+    Probabilities are loaded from csv file.
+    :param csv_filename: csv fixture from tests.fixtures.filenames.csv_filename
+    :return: None
+    """
+    input_features = [
+        text_feature(vocab_size=10, min_len=1, representation='sparse'),
+        category_feature(vocab_size=10)
+    ]
+    output_features = [category_feature(vocab_size=2, reduce_input='sum')]
+
+    # Generate test data
+    rel_path = generate_data(input_features, output_features, csv_filename)
+    input_features[0]['encoder'] = 'parallel_cnn'
+
+    config = {
+        'input_features': input_features,
+        'output_features': output_features,
+        'combiner': {
+            'type': 'concat',
+            'fc_size': 14
+        },
+        'training': {'epochs': 2}
+    }
+
+    model = LudwigModel(
+        config,
+        logging_level=logging.WARN
+    )
+
+    (
+        _,
+        preprocessed_data,
+        exp_dir_name
+    ) = model.train(dataset=rel_path)
+
+    _, _, test_set, metadata = preprocessed_data
+
+    model.predict(
+        dataset=test_set,
+        skip_save_predictions=False,
+        output_directory=exp_dir_name
+    )
+
+    output_feature_name = config['output_features'][0][NAME]
+    probability = os.path.join(exp_dir_name, '{}_probabilities.csv').format(
+        output_feature_name)
+    experiment_source_data_name = csv_filename.split('.')[0]
+    split_file = get_split_path(csv_filename)
+
+    ground_truth = _extract_ground_truth_values(
+        rel_path,
+        output_feature_name,
+        ground_truth_split=2,
+        split_file=split_file
+    )
```

Comment:
not needed, you already have the test set from the output of training

Reply:
@w4nderlust   I'd like to discuss this item.

Based on an earlier discussion, we talked about the viz function would be the place to translate the categories raw string values to the encoded numeric values.  See highlighted section.
![image](https://user-images.githubusercontent.com/1425269/98444477-e19eab00-20df-11eb-8b65-8285688ef8a2.png)

This current method works with how the _cli function works.   

So to make the viz function support both the api and cli test functions, I'll have to modify the cli test function to do the translation of categorical raw string values to the encoded numeric value.

I just want to confirm that this is OK because, if I recall correctly, the location for this function has changed several times between  within the cli or within the viz functions.

Reply:
i see, you are probably right, before the data outputted by train was unprocessed, now it is processed. So, what i propose to do is the following: in the viz (as in non cli) function i will add a check like:
```
if not isinstance(ground_truth, Dataset):
  # do the mapping string to int
else:
  # it's already ints
```
So the type for `ground_truth` should be `Union[pd.Series, Dataset]` or something like that.
Does ths make sense? Sorry for the back and forth about this issue, but I'm also figuring out the best API as we go :) But I think with this change the viz function will become waaay more useable.

Reply:
Implemented the recommended test